guides/source/asset_pipeline.md

```diff
@@ -8,479 +8,616 @@ This guide covers the asset pipeline.
-Pinning "object-assign" to https://ga.jspm.io/npm:object-assign@4.1.1/index.js
-```
+When fingerprinted, a file like `styles.css` is renamed to
+`styles-a1b2c3d4e5f6.css`. This ensures that if `styles.css` is updated, the
+filename changes as well, compelling the browser to download the latest version
+instead of using a potentially outdated cached copy.
 
-You can also pick an alternative, supported CDN provider when pinning, like [`unpkg`](https://unpkg.com/) or [`jsdelivr`](https://www.jsdelivr.com/) ([`jspm`](https://jspm.org/) is the default):
+A manifest is a file that maps logical asset paths to their corresponding
+fingerprinted filenames. The manifest is used to look up the correct filename
+for an asset. The manifest is typically a JSON file that looks like this:
 
-```bash
-$ bin/importmap pin react --from jsdelivr
-Pinning "react" to https://cdn.jsdelivr.net/npm/react@17.0.2/index.js
+```json
+{
+  "application.js": "application-1a2b3c4d5e6f.js",
+  "application.css": "application-1a2b3c4d5e6f.css"
+}
 ```
 
-Remember, though, that if you switch a pin from one provider to another, you may have to clean up dependencies added by the first provider that isn't used by the second provider.
-
-Run `bin/importmap` to see all options.
-
-Note that this command is merely a convenience wrapper to resolving logical package names to CDN URLs. You can also just lookup the CDN URLs yourself, and then pin those. For example, if you wanted to use Skypack for React, you could just add the following to `config/importmap.rb`:
+Fingerprinting is enabled by default for both the development and production
+environments. When you reference an asset in your application, Rails uses the
+manifest to find the correct fingerprinted filename to include in the HTML
+output.
+
+Features
+--------
+
+Propshaft expects that your assets are already in a browser-ready format—like
+plain CSS, JavaScript, or preprocessed images (like JPEGs or PNGs). Propshaft’s
+job is to organize, version, and serve those assets efficiently. If you want to
+use advanced features, like [Sass](https://sass-lang.com/) for styles or modern
+JavaScript features, you’d usually handle those with a separate tool (like
+[Webpack](https://webpack.js.org/), [Vite](https://vite.dev/), or
+[esbuild](https://esbuild.github.io/)) alongside Propshaft.
+
+In this section, we’ll cover the main features of Propshaft, how it works and
+compare it with its predecessor, [Sprockets](https://github.com/rails/sprockets-rails).
+
+### No dependency chains
+
+In Sprockets, you could link files
+together so they were loaded in the correct order. For example, you may have had
+a main JavaScript file that relied on other files to be loaded first. Sprockets
+managed this for you automatically, checking dependencies and loading everything
+in the correct order.
+
+Propshaft doesn’t automatically understand these dependencies. If
+you have files that depend on each other, you’ll need to manage the order in
```

Comment:
Should this be called out like a warning or something?

Reply:
I ended up moving sections around and updated the phrasing to 

"With Propshaft, you can control the loading order of dependent files by
organizing them manually or ensuring they are included in the correct sequence
within your HTML or layout files...."

Hence, this won't apply any longer.